editor/language_server/src/tests/mocks.rs

```diff
@@ -100,6 +100,19 @@ impl MockService {
         self.get_diagnostics().await
     }
 
+    pub async fn completion(
+        &mut self,
+        uri: &Url,
+        position: Position,
+    ) -> Option<CompletionResponse> {
+        self.assert_ready();
+        let hover_resp = self
+            .send(text_document::definition::request(uri, position))
+            .await
+            .unwrap();
+        serde_json::from_value(hover_resp.get("result").unwrap().clone()).ok()
```

Comment:
so if you can serialized to json, then this is valid. I'm not sur eI understand what this is doing.

Reply:
hover_resp is json, we get the result key, then ask to convert that from json to a custom type